### ISSUE TITLE
[Security] v2.10.2 contains 35 fixable vulnerabilities

### ISSUE THREAD

**Issue by hpvd** [apache/pulsar#18348]
### Search before asking

- [X] I searched in the [issues](https://github.com/apache/pulsar/issues) and found nothing similar.

### Version

v2.10.2

### Minimal reproduce step

look into trivy powered inspection for vulnerabilities
at artifacthub.io
https://artifacthub.io/packages/helm/apache/pulsar?modal=security-report

open details of in the latest helm chart v3.0.0 included pulsar v2.10.2 image

### What did you expect to see?

very few fixable vulnerabilities, since v2.10.2 was released just 8 days ago https://github.com/apache/pulsar/releases

### What did you see instead?

- 72 vulnerabilities have been detected in the image
- 35 of these should be fixable with a version bump of dependencies.

![2022-11-04_17h06_11](https://user-images.githubusercontent.com/5681880/200025536-a6808fb3-2f4a-4e31-92ba-9c5f61ebe4f1.png)

![2022-11-04_17h03_17](https://user-images.githubusercontent.com/5681880/200025573-871fc438-6d24-4788-90c9-cce8bf7ab477.png)

### Anything else?

- this is related to
https://github.com/apache/pulsar/issues/18338
- this is a follow up of
https://github.com/apache/pulsar/issues/18041
- this is part of
https://github.com/apache/pulsar-helm-chart/issues/334

### Are you willing to submit a PR?

- [ ] I'm willing to submit a PR!

**Comment by michaeljmarshall** [comment]
Several of these have been fixed, and others are brought in by presto. @nicoloboschi does security scanning of the pulsar code base and frequently contributes updates to dependencies, when possible.

**Comment by hpvd** [comment]
yes and there is an automated OWASP Dependency Check running since June 2021 https://github.com/apache/pulsar/pull/10855

doing all this,
our current procedure/routine seems to miss these 35 fixable vulnerabilities (CVE) when releasing latest version 2.10.2
okay a (very) few less... see https://github.com/apache/pulsar/issues/8815#issuecomment-1303875392

**Comment by Jason918** [comment]
> -> but that's the state of our latest, only 8 days old release.
> not sooo bad, but not so good either.

The final release candidate of 2.10.2 is 23 days ago. And most of fixes comes in the early September.

I think the problem is that the current release process have too many work and not automated. It took too long for a version to be released. And I don't see any check process of CVEs before we cut the release. I am pretty sure "OWASP Dependency Check" passed before the first release candidate is cut.

**Comment by hpvd** [comment]
Looks like there are several points producing this non-perfect state:
- I absolutely agree with @Jason918 that's probably part of the problem, that the time from last fixes to final release may be too long
- do we have already a named point in the release check list that applying all (as many as possible) available fixes before release is a must (quality gate, similar to https://github.com/apache/pulsar/issues/8815)?
- another part may be the manual and not time constraint suppression of positive findings (for what ever reason) like named here https://github.com/apache/pulsar/issues/17068
- what else do we have to consider?

**Comment by michaeljmarshall** [comment]
> I think the problem is that the current release process have too many work and not automated. It took too long for a version to be released.

I agree. I think we're moving in the right direction by breaking out the C++ and Python clients, but it will take time to see those benefits. Maybe it is worth a mailing list discussion of how to share the load on the release manager?

**Comment by Jason918** [comment]
> > I think the problem is that the current release process have too many work and not automated. It took too long for a version to be released.
> 
> I agree. I think we're moving in the right direction by breaking out the C++ and Python clients, but it will take time to see those benefits. Maybe it is worth a mailing list discussion of how to share the load on the release manager?

Yes, this is one of the motivations of [PIP-175: Extend time based release process ](https://github.com/apache/pulsar/issues/15966)

**Comment by hpvd** [comment]
when looking through the security report again,
I was wondering why there is netty version 3.10.6 listet with 3 named vulnerabilities.
This is a version from the year 2016 (https://mvnrepository.com/artifact/io.netty/netty/3.10.6.Final)
and as far I have seen, we have upgraded to >netty 4.1

the question:
Can it also be a part of the problem that we only upgrade "some points of use" in the code when there is a notification about a security issue?

**Comment by hpvd** [comment]
just add some release dates of the versions with fixes:
![2022-11-11_16h19_131](https://user-images.githubusercontent.com/5681880/201386827-d3b9ea86-884e-4bed-9612-36742cc012b9.png)

**Comment by hpvd** [comment]
hmm just another number: the ubuntu included, added 77 vulnerabilities with CVE...

**Comment by Jason918** [comment]
> I was wondering why there is netty version 3.10.6 listet with 3 named vulnerabilities.

This is from `presto-sql/pulsar-presto-distribution`. `presto-sql` contains a lot of legacy dependencies and not easily updated. Previously, I saw @tisonkun working on some updates to Trino in PulsarSQL. Not sure if it's related.


@hpvd I think you can start a discuss thread on the mail-list to get more attention on this matter.

**Comment by tisonkun** [comment]
@Jason918 No. The upgrade of Trino (PrestoSQL) dependency happened only for the unreleased master (2.12.0).

**Comment by tisonkun** [comment]
And yes, if we don't update Trino dependency for 2.10 (I think so), these reports should be suppressed.

**Comment by compuguy** [comment]
This issue is still relevant with release 3.1.0. Plenty of vulnerabilities in the latest release, especially with the connectors and offloaders...

**Comment by hpvd** [comment]
when stale label is added automatically,
it should also be removed automatically if an issue is newly commented or maybe even if it's mentioned some else.
